tidy3d/web/webapi.py

```diff
@@ -240,40 +235,58 @@ def monitor(task_id: TaskId) -> None:
 
     break_statuses = ("success", "error", "diverged", "deleted", "draft")
 
-    def get_status():
-        """Get status for this task (called many times below, so put into function)."""
-        status = get_info(task_id).status
+    show_cost = True
+
+    def get_status(show_cost=False):
+        """Get status for this task (called many times below, so put into function).
+        If the estimated cost is available, it is displayed if requested."""
+        task_info = get_info(task_id)
+        status = task_info.status
         if status == "visualize":
             return "success"
-        return status
+        if status == "error":
+            raise WebError("Error running task!")
+
+        # log the maximum flex unit cost
+        est_flex_unit = task_info.estFlexUnit
+        if show_cost is True and est_flex_unit is not None and est_flex_unit > 0:
+            log.info(f"Maximum flex unit cost: {est_flex_unit:1.2f}")
+            # Set show_cost to False so that it is only shown once during the run
+            show_cost = False
+
+        return status, show_cost
 
     console = Console()
 
+    status, show_cost = get_status(show_cost=show_cost)
+    log.info(f"status = {status}")
     # already done
-    if get_status() in break_statuses:
-        log.info(f"status = {get_status()}")
+    if status in break_statuses:
         return
 
     # preprocessing
     with console.status(f"[bold green]Starting '{task_name}'...", spinner="runner"):
-        while get_status() not in break_statuses and get_status() != "running":
-            if status != get_status():
-                status = get_status()
+        while status not in break_statuses and status != "running":
```

Comment:
I think this can be dangerous, which is why I intetially call get_status in the whole statement.  if line 272 never gets run to change `status`, it could get stuck in an infinite while loop?  same below on line 279

Reply:
So `status` is originally something that can enter the while loop. Then on line 270 we call `get_status` and change `status` if we got something new. So if `status` never changes it's not meant to exit the loop because `status` is still not a breaking status or `running`?